src/components/TodoItem.tsx

```diff
@@ -0,0 +1,262 @@
+          });
+        } finally {
+          setIsLoading(false);
+        }
+
+        return;
+      }
+
+      await updateTodo(updatedTodo);
+
+      dispatch({
+        type: 'updateTodo',
+        payload: { todo: updatedTodo },
+      });
+      setIsUpdatingTitle(false);
+    } catch (error) {
+      if (isUpdatingTitle && updateTitleRef.current) {
+        updateTitleRef.current.focus();
+      }
+
+      dispatch({
+        type: 'error',
+        payload: { error: ErrorMessage.Updating },
+      });
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handleKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Escape') {
+      setIsUpdatingTitle(false);
+      setNewTitle(title);
+    }
+  };
+
+  useEffect(() => {
+    switch (true) {
+      case shouldLoading === LoadingStatus.Completed && completed:
+        setIsLoading(true);
+
+        break;
+
+      case shouldLoading === LoadingStatus.All:
+        setIsLoading(true);
+
+        break;
+
+      case shouldLoading === LoadingStatus.Current && id === TEMP_TODO_ID:
+        setIsLoading(true);
+
+        break;
+
+      case shouldLoading === LoadingStatus.None:
+        setIsLoading(false);
+
+        break;
+
+      default:
+        break;
+    }
+  }, [id, completed, shouldLoading]);
+
+  useEffect(() => {
+    if (isUpdatingTitle) {
+      updateTitleRef.current?.focus();
+    }
+  }, [isUpdatingTitle]);
+
+  return (
+    <div
+      data-cy="Todo"
+      className={cn(
+        'todo',
+        { completed },
+      )}
+      key={id}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={completed}
+          onChange={toggleChecked}
+        />
+      </label>
+
+      {
+        !isUpdatingTitle
+          ? (
+            <>
+              <span
+                data-cy="TodoTitle"
+                className="todo__title"
+                onDoubleClick={() => {
+                  setIsUpdatingTitle(true);
+                }}
+              >
+                {todo.title}
+              </span>
+
+              <button
+                type="button"
+                className="todo__remove"
+                data-cy="TodoDelete"
+                onClick={deleteCurrentTodo}
+              >
+                ×
+              </button>
+            </>
+          )
+          : (
+            <form
+              onSubmit={e => {
+                e.preventDefault();
+                handleNewTitleSubmit();
+              }}
+            >
+              <input
+                data-cy="TodoTitleField"
+                type="text"
+                className="todo__title-field"
+                placeholder="Empty todo will be deleted"
+                value={newTitle}
+                onChange={e => setNewTitle(e.target.value)}
+                onBlur={handleNewTitleSubmit}
```

Comment:
we shouldn't submit value on blur. Since it means that user'd like not to save new data (the same as `escape`). Save data only when user submit new title

Reply:
I am a little confused because of the requirements we have:
- saves changes on the form submit (just press `Enter`);
- save changes when the field loses focus (`onBlur`);

Please tell me what I should do?